### PR TITLE
Add k8s-staging-apisnoop project for images

### PR DESCRIPTION
**Pull request by hh**
/assign @justaugustus 
/wip

Needing to automate builds for APISnoop images for use with prow.

### CODE REVIEW

**Comment by hh** [comment]
I'm currently using the apisnoop gcs project, which I created.
It's unclear to me who creates the k8s-staging-apisnoop project, or if we just refer to the existing one.

I'm also unclear what is required in groups/groups.yaml for apisnoop, since that seems to affect who can push to the staging repo.

File: k8s.gcr.io/images/k8s-staging-apisnoop/images.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# These images do not exist yet
+- name: auditlogger
+  dmap:
+    "sha256:46b237764fee52288a7ab075353c77d4e0de55e374c87474409091bd94e442af": ["v1.0.0"]
```
Comment (justaugustus) [inline k8s.gcr.io/images/k8s-staging-apisnoop/images.yaml]: Let's drop the SHAs until images actually exist.

File: k8s.gcr.io/images/k8s-staging-apisnoop/OWNERS [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+- hh
+- bobymcbobs
```
Comment (justaugustus) [inline k8s.gcr.io/images/k8s-staging-apisnoop/OWNERS]: @bobymcbobs would need to be an org member to do reviews/approvals for apisnoop --> https://github.com/kubernetes/org/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.md&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E

**Comment by justaugustus** [comment]
> I'm currently using the apisnoop gcs project, which I created.
> It's unclear to me who creates the k8s-staging-apisnoop project, or if we just refer to the existing one.

I'm fairly sure that the project would need to be on K8s Infra for everything to be properly configured.
`ensure-staging-projects.sh` is run by someone with admin access across our GCP properties to create/reconcile the project, buckets, registries, retention settings, and IAM.

I _believe_ Christoph and Dims are the only ones with that access at the present moment and that happens near merge time, post-approvals.

> I'm also unclear what is required in groups/groups.yaml for apisnoop, since that seems to affect who can push to the staging repo.

Yep, membership to the Google Group will grant IAM over the GCP project of the same name once the reconcile script is run.

File: groups/groups.yaml [hunk 3]
```diff
@@ -743,6 +743,20 @@ groups:
...
 
+  - email-id: k8s-infra-apisnoop@kubernetes.io
+    name: k8s-infra-apisnoop
+    description: |-
+      ACL for apisnoop images
+    settings:
+      WhoCanPostMessage: "ANYONE_CAN_POST"
+      ReconcileMembers: "true"
+    members:
+      - caleb@ii.coop
```
[the remark this answers is not in the document]
Comment (hh) [reply groups/groups.yaml]: :heavy_check_mark:

File: infra/gcp/ensure-staging-storage.sh [hunk 4]
```diff
@@ -39,6 +39,7 @@ function usage() {
 
 # NB: Please keep this sorted.
 STAGING_PROJECTS=(
+    apisnoop
```
[the remark this answers is not in the document]
Comment (hh) [reply infra/gcp/ensure-staging-storage.sh]: :heavy_check_mark:

File: k8s.gcr.io/images/k8s-staging-apisnoop/images.yaml (hunk 1 above)
Comment (hh) [reply k8s.gcr.io/images/k8s-staging-apisnoop/images.yaml]: :heavy_check_mark: 
Removed, left a placeholder comment.

File: k8s.gcr.io/images/k8s-staging-apisnoop/OWNERS (hunk 2 above)
Comment (hh) [reply k8s.gcr.io/images/k8s-staging-apisnoop/OWNERS]: Looks like @BobyMCbobs is already an org member.

File: groups/groups.yaml [hunk 5]
```diff
@@ -743,6 +743,20 @@ groups:
       - thockin@google.com
       - bartek@smykla.com
 
+  - email-id: k8s-infra-apisnoop@kubernetes.io
```
[the remark this answers is not in the document]
Comment (dims) [reply groups/groups.yaml]: /hold

this suggestion has not been fixed yet @hh

[the remark this answers is not in the document]
Comment (justaugustus) [reply groups/groups.yaml]: @dims Looks to be an outdated diff. Seems fine in the full file view.

[the remark this answers is not in the document]
Comment (dims) [reply groups/groups.yaml]: @justaugustus the name and email id does not match yet.

[the remark this answers is not in the document]
Comment (justaugustus) [reply groups/groups.yaml]: I need coffee :man_facepalming:

[the remark this answers is not in the document]
Comment (bartsmykla) [reply groups/groups.yaml]: It looks like it has been fixed by @hh now. I'm accepting the PR.

**Comment by dims** [comment]
next time, please don't cancel `/hold` until one of us are able to run the scripts for the email and staging repo creation :) will do this in another hour or so after meetings.

**Comment by justaugustus** [comment]
@dims -- Roger that. Thought the hold was just for the groups.yaml fix.

**Comment by dims** [comment]
DONE 

- mailing list created
- staging gcs repo created

**Comment by hh** [comment]
Thanks @dims!

**Pull request merged**